test/test_type.jl

```diff
@@ -497,6 +513,158 @@ isdir(outdir) && rm(outdir, recursive = true)
+                  RealT
+            @test eltype(@inferred Trixi.max_abs_speeds(u, equations)) == RealT
+            @test eltype(@inferred cons2prim(u, equations)) == RealT
+            @test eltype(@inferred prim2cons(u, equations)) == RealT
+            @test eltype(@inferred cons2entropy(u, equations)) == RealT
+            @test eltype(@inferred entropy2cons(u, equations)) == RealT
+            @test eltype(@inferred Trixi.total_entropy(u, equations)) == RealT
+            @test eltype(@inferred Trixi.temperature(u, equations)) == RealT
+            @test eltype(@inferred Trixi.totalgamma(u, equations)) == RealT
+            @test eltype(@inferred density(u, equations)) == RealT
+            @test eltype(@inferred pressure(u, equations)) == RealT
+        end
+    end
+
+    @timed_testset "Compressible Euler Multicomponent 2D" begin
+        for RealT in (Float32, Float64)
+            gammas = (RealT(1.4), RealT(1.4))
+            gas_constants = (RealT(0.4), RealT(0.4))
+            equations = @inferred CompressibleEulerMulticomponentEquations2D(gammas = gammas,
+                                                                             gas_constants = gas_constants)
+
+            x = SVector(zero(RealT), zero(RealT))
+            t = zero(RealT)
+            u = u_ll = u_rr = SVector(one(RealT), one(RealT), one(RealT), one(RealT),
+                                      one(RealT))
+            orientations = [1, 2]
+            normal_direction = SVector(one(RealT), zero(RealT))
+
+            @test eltype(@inferred initial_condition_convergence_test(x, t, equations)) ==
+                  RealT
+            @test eltype(@inferred initial_condition_weak_blast_wave(x, t, equations)) ==
+                  RealT
+
+            @test eltype(@inferred source_terms_convergence_test(u, x, t, equations)) ==
+                  RealT
+
+            @test eltype(@inferred flux(u, normal_direction, equations)) == RealT
+            if RealT == Float32
+                # check `ln_mean` (test broken)
+                @test_broken eltype(@inferred flux_ranocha(u_ll, u_rr, normal_direction,
+                                                           equations)) == RealT
+            else
+                @test eltype(@inferred flux_ranocha(u_ll, u_rr, normal_direction,
+                                                    equations)) == RealT
+            end
+
+            for orientation in orientations
+                @test eltype(@inferred flux(u, orientation, equations)) == RealT
+                # pass with check for `ln_mean` 
+                @test eltype(@inferred flux_chandrashekar(u_ll, u_rr, orientation,
+                                                          equations)) ==
+                      RealT
+                # pass with check for `ln_mean` and `inv_ln_mean`
+                @test eltype(@inferred flux_ranocha(u_ll, u_rr, orientation, equations)) ==
+                      RealT
```

Comment:
I'm wondering why the tests with `orientation` pass but the ones above with `normal_direction` work only for `Float64`. Both should use some `ln_mean`, don't they?

Reply:
Yes the same `ln_mean`.

Reply:
https://github.com/trixi-framework/Trixi.jl/blob/2043cfe3f79de2b77124c13d57de96a8ddf0ea52/src/equations/compressible_euler_multicomponent_2d.jl#L439-L466

Reply:
With `normal_direction`
https://github.com/trixi-framework/Trixi.jl/blob/2043cfe3f79de2b77124c13d57de96a8ddf0ea52/src/equations/compressible_euler_multicomponent_2d.jl#L520-L536

Reply:
You can see that `f_rho` is okay for both and the problem is actually the `f_other`, which is specified to type `real(equations)` with `orientation` but not with `normal_direction`.

Btw, this coding style does not look good to me (and also not good to the repo) if both function come from the same person. At least keep the similar functions in a similar coding way.

I think maybe keep both with `real(equations)` or both without `real(equations)` if no specific reasons provided for doing so. How do you think @ranocha

Reply:
You're right - `f_other = SVector{3, real(equations)}(f1, f2, f3)` should just be `f_other = SVector(f1, f2, f3)`.